WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/images/ai/CreateInputImageFromUrl.kt

```
@@ -0,0 +1,37 @@
+package com.woocommerce.android.ui.products.images.ai
+
+import android.content.Context
+import androidx.core.graphics.drawable.toBitmap
+import coil.ImageLoader
+import coil.request.ImageRequest
+import coil.request.SuccessResult
+import com.google.mlkit.vision.common.InputImage
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class CreateInputImageFromUrl @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+    suspend operator fun invoke(imageUrl: String): Result<InputImage> {
+        return try {
+            val imageLoader = ImageLoader(context)
+            val imageRequest = ImageRequest.Builder(context)
+                .data(imageUrl)
+                .build()
+
+            val result = imageLoader.execute(imageRequest)
+            when (result) {
+                is SuccessResult -> {
+                    val bitmap = result.drawable.toBitmap()
+                    val inputImage = InputImage.fromBitmap(bitmap, 0)
+                    Result.success(inputImage)
+                }
+                else -> Result.failure(Exception("Image loading failed"))
+            }
+        } catch (e: IllegalArgumentException) {
+            Result.failure(e)
+        } catch (e: OutOfMemoryError) {
+            Result.failure(Exception("Out of memory loading image", e))
```

❓ Is out-of-memory something the app can recover from? Feels a bit unconventional to return a Result instead of re-throwing the exception. I'd be even worried the JVM's memory might be in a corrupted state in this case, so it might be unsafe to recover.

We could potentially try to check whether the image will fit the memory before creating the bitmap. Wdyt?

---

Good point. I updated the coil `ImageRequest` to resize the image to `2048 × 2048`, which should result in ~16.7 MB memory usage, which I think is acceptable. Also stopped swallowing the `OOM` exception: 34ba6ed8a9e74f458f785b8d9c091f67977662a9